addons/pos_adyen/models/adyen_account.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import uuid
+from werkzeug.urls import url_join
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AdyenAccount(models.Model):
+    _inherit = 'adyen.account'
+
+    store_ids = fields.One2many('adyen.store', 'adyen_account_id')
+    terminal_ids = fields.One2many('adyen.terminal', 'adyen_account_id')
+
+    @api.model
+    def create(self, values):
+        store_ids = values.pop('store_ids', [])
+        adyen_account_id = super(AdyenAccount, self).create(values)
+        adyen_account_id.write({'store_ids': store_ids})
+        return adyen_account_id
+
+    def action_order_terminal(self):
+        if not self.store_ids:
+            raise ValidationError(_('Please create a store first.'))
+
+        store_uuids = ','.join(self.store_ids.mapped('store_uuid'))
+        onboarding_url = self.env['ir.config_parameter'].sudo().get_param('adyen_marketpay.onboarding_url')
+        return {
+            'type': 'ir.actions.act_url',
+            'target': 'new',
+            'url': url_join(onboarding_url, 'order_terminals?store_uuids=%s' % store_uuids),
+        }
+
+
+class AdyenStore(models.Model):
```

Comment:
Is there any validation status  or something like that on the stores?

Reply:
Stores only have a status that is `pending`, `active` or `closed`. It switches automatically from `pending` to `active` once it's created in Adyen's DB, which takes a few seconds. The `closed` is set manually when a store is deleted. So I don't think we need to sync their status.

Reply:
ok cool